File: 백준/Bronze/2609. 최대공약수와 최소공배수/최대공약수와 최소공배수.py
import sys

input = sys.stdin.readline
a,b = map(int,input().split())


def gcd(a,b):
    if a > b:
        while b:
            a,b = b, a%b
        return a
    elif b > a:
        while a:
            b,a = a, b%a
        return b
    else:
        return a

def lcd(a,b):
    return a*b//gcd(a,b)



print(gcd(a,b))
print(lcd(a,b))

File: 프로그래머스/1/12940. 최대공약수와 최소공배수/최대공약수와 최소공배수.js
function solution(n, m) {
    var answer = [];
    function gcd(a,b){
        while (b !== 0){
            let tmp = b
            b = a%b
            a = tmp
        }
        return a
    }
    if (n > m) {
        let g = gcd(n,m)
        let lcd = n*m/g
        answer.push(g,lcd)
        

    }
    else {
        let g = gcd(m,n)
        let lcd = n*m/g
        answer.push(g,lcd)
       
    }
    
    return answer;
}